include/range/v3/view/exclusive_scan.hpp

```diff
@@ -0,0 +1,164 @@
+                : view_adaptor<exclusive_scan_view<Rng, T, Fun>, Rng> {
+        private:
+            friend range_access;
+            T init_;
+            semiregular_t<Fun> fun_;
+            using single_pass = SinglePass<iterator_t<Rng>>;
+            using use_sentinel_t = meta::or_<meta::not_<BoundedRange<Rng>>, single_pass>;
+
+            template<bool IsConst>
+            struct adaptor : adaptor_base {
+            private:
+                using exclusive_scan_view_t = meta::const_if_c<IsConst, exclusive_scan_view>;
+                // sum_'s type doesn't have to be range_value_type_t<Rng>.
+                semiregular_t<range_value_type_t<Rng>> sum_;
+                exclusive_scan_view_t *rng_;
+            public:
+                using single_pass = exclusive_scan_view::single_pass;
+                adaptor() = default;
+                adaptor(exclusive_scan_view_t &rng)
+                        : rng_(&rng)
+                {}
+                iterator_t<Rng> begin(exclusive_scan_view_t &)
+                {
+                    sum_ = rng_->init_;
+                    return ranges::begin(rng_->base());
+                }
+                range_value_type_t<Rng> read(iterator_t<Rng>) const
+                {
+                    return sum_;
+                }
+                void next(iterator_t<Rng> &it)
+                {
+                    if (it != ranges::end(rng_->base()))
+                    {
+                        auto &current = static_cast<range_value_type_t<Rng> &>(sum_);
+                        sum_ = invoke(rng_->fun_, current, *it);
+                        ++it;
+                    }
+                }
+                void prev() = delete;
+            };
+
+            adaptor<false> begin_adaptor()
+            {
+                return {*this};
+            }
+            meta::if_<use_sentinel_t, adaptor_base, adaptor<false>> end_adaptor()
+            {
+                return {*this};
+            }
+            CONCEPT_REQUIRES(Range<Rng const>() &&
+                             Invocable<Fun const&, range_common_reference_t<Rng>,
+                                     range_common_reference_t<Rng>>())
+            adaptor<true> begin_adaptor() const
+            {
+                return {*this};
+            }
+            CONCEPT_REQUIRES(Range<Rng const>() &&
+                             Invocable<Fun const&, range_common_reference_t<Rng>,
+                                     range_common_reference_t<Rng>>())
+            meta::if_<use_sentinel_t, adaptor_base, adaptor<true>> end_adaptor() const
+            {
+                return {*this};
+            }
+
+        public:
+            exclusive_scan_view() = default;
+            exclusive_scan_view(Rng rng, T init, Fun fun)
+            : exclusive_scan_view::view_adaptor{std::move(rng)}
+            , init_(std::move(init))
+            , fun_(std::move(fun))
+            {}
+            CONCEPT_REQUIRES(SizedRange<Rng>())
```

Comment:
This applies `size` to a `const` range: it requires `SizedRange<Rng const>`. It wouldn't hurt to have both a `const` and non-`const` overload of `size`, since some underlying views will only provide a non-`const` `size`.